### PR TITLE
Added DeleteUserByUserId() method

### PR DESCRIPTION
**Pull request by M-adform**
Mes pilnai negalime trinti userio, todėl metodas yra put ir keičia parametrą is_Deleted.

### CODE REVIEW

File: project_backend/Controllers/AdminController.cs [hunk 1]
```diff
@@ -13,22 +13,32 @@ public class AdminController(IAdminService adminService) : ControllerBase
         [HttpGet]
         public async Task<IActionResult> GetUsersAsync()
         {
-            try
+            var usersList = await _adminService.GetUsersAsync();
+
+            if (usersList == null)
```
Comment (jplau123) [inline project_backend/Controllers/AdminController.cs:18]: rekomenduoju naudoti ?? operatoriu:
``` var usersList = await _adminService.GetUsersAsync() ?? throw new NotFoundException("No users found."); ```

File: project_backend/Repositories/AdminRepository.cs [hunk 2]
```diff
@@ -11,9 +11,32 @@ public class AdminRepository(IDbConnection dbConnection) : IAdminRepository
...
         {
-            string query = "SELECT * FROM users;";
+            string query = "SELECT * FROM users WHERE is_deleted = false";
             var result = await _connectionString.QueryAsync<User>(query);
             return result.ToList();
         }
+
+        public async Task<User> GetUserByUserIdAsync(int id)
+        {
+            string query = "SELECT * FROM users WHERE user_id = @id";
```
Comment (jplau123) [inline project_backend/Repositories/AdminRepository.cs]: Geriau nenaudoti * operatoriaus, o isvardinti visus stulpelius, kuriu reikia.

File: project_backend/Scripts/Script 0001 - Create tables.sql [hunk 3]
```diff
@@ -8,7 +8,8 @@ active bool default true,
 created_at timestamp default current_timestamp,
 created_by varchar(255),
 modified_at timestamp,
-modified_by varchar(255)
+modified_by varchar(255),
+is_deleted boolean
```
Comment (jplau123) [inline project_backend/Scripts/Script 0001 - Create tables.sql:12]: cia geriau uzdet "DEFAULT FALSE"

File: project_backend/Middlewares/ErrorMiddleware.cs [hunk 4]
```diff
@@ -25,13 +26,20 @@ public async Task InvokeAsync(HttpContext httpContext)
...
+                trace = ex.StackTrace;
+            }
             catch (Exception ex)
             {
                 statusCode = (int)HttpStatusCode.InternalServerError;
                 message = $"{ex.Message}";
                 trace = ex.StackTrace;
             }
 
+
```
Comment (uadform) [inline project_backend/Middlewares/ErrorMiddleware.cs]: kam cia extra eilute?

File: project_backend/Services/AdminService.cs [hunk 5]
```diff
@@ -12,5 +12,22 @@ public class AdminService(IAdminRepository adminRepository) : IAdminService
             var departmentList = await _adminRepository.GetUsersAsync();
             return departmentList;
         }
+
+        public async Task<int> DeleteUserByUserIdAsync(int id)
+        {
+            var user = await _adminRepository.GetUserByUserIdAsync(id);
+
+            if (user.Is_Deleted)
```
Comment (jplau123) [inline project_backend/Services/AdminService.cs]: cia pataisyt kaip taremes irl

File: project_backend/Services/UserService.cs [hunk 6]
```diff
@@ -13,5 +15,28 @@ public int AddUserBalance(int user_id, int balance)
...
+        {
+            var result = await _userRepository.GetUsersAsync();
+            return result.ToList();
+        }
+
+        public async Task DeleteUserByUserIdAsync(int id)
+        {
+            var user = await _userRepository.GetUserByUserIdAsync(id);
+
+            if (user == null)
```
Comment (jplau123) [inline project_backend/Services/UserService.cs:29]: Siulyciau rasyti ``` var user = await _userRepository.GetUserByUserIdAsync(id) ?? throw new NotFoundException("Could not find user with provided id.");```

**Review by uadform — APPROVED** [review]
**Good job**

**Review by jplau123 — APPROVED** [review]
All good.

File: project_backend/Repositories/UserRepository.cs [hunk 7]
```diff
@@ -22,10 +23,37 @@ public int AddUserBalance(int user_id, int balance)
...
+            {
+                id
+            };
+
+            return await _connection.QuerySingleAsync<User>(query, queryArguments);
+        }
+
+        public async Task<int> DeleteUserByUserIdAsync(int id)
+        {
+            string query = "UPDATE users SET is_deleted = @isDeleted WHERE user_id = @id";
```
Comment (Gabriele-ele) [inline project_backend/Repositories/UserRepository.cs:49]: gal irgi reiketu @is_deleted?

**Review by Gabriele-ele — APPROVED** [review]

**Pull request merged**